Modules/IO/IPL/include/itkIPLFileNameList.h

```diff
@@ -93,21 +93,15 @@ class IPLFileSortInfo
 
   virtual ~IPLFileSortInfo();
 
-  IPLSetMacroDeclaration(ImageFileName, std::string);
-  IPLGetMacroDeclaration(ImageFileName, std::string);
-  IPLSetMacroDeclaration(SliceLocation, float);
-  IPLGetMacroDeclaration(SliceLocation, float);
-  IPLSetMacroDeclaration(SliceOffset, int);
-  IPLGetMacroDeclaration(SliceOffset, int);
-  IPLSetMacroDeclaration(EchoNumber, int);
-  IPLGetMacroDeclaration(EchoNumber, int);
-  IPLSetMacroDeclaration(ImageNumber, int);
-  IPLGetMacroDeclaration(ImageNumber, int);
-  IPLSetMacroDeclaration(Data, void *);
-  IPLGetMacroDeclaration(Data, const void *);
+  IPLSetMacroDeclaration(ImageFileName, std::string)
+  IPLGetMacroDeclaration(ImageFileName, std::string) IPLSetMacroDeclaration(SliceLocation, float)
+  IPLGetMacroDeclaration(SliceLocation, float) IPLSetMacroDeclaration(SliceOffset, int)
+  IPLGetMacroDeclaration(SliceOffset, int) IPLSetMacroDeclaration(EchoNumber, int)
+  IPLGetMacroDeclaration(EchoNumber, int) IPLSetMacroDeclaration(ImageNumber, int)
+  IPLGetMacroDeclaration(ImageNumber, int) IPLSetMacroDeclaration(Data, void *)
+  IPLGetMacroDeclaration(Data, const void *)
 
-private:
-  std::string  m_ImageFileName{};
+    private : std::string m_ImageFileName{};
```

Comment:
Formatting wrong here too. Hopefully running clang-format will fix it, since you added these macros to `StatementMacros`.

Reply:
I would really prefer to NOT remove the semi-colons.  It makes clang-formatting very challenging and tedious.  Additionally, many modern IDE's complain about the lack of ; at the end of what is interpreted as a statement.

I would prefer that the macros are always terminated by a statement, or a NOOP statement as is proposed in #4706

Reply:
>It makes clang-formatting very challenging and tedious.

How so?  Are you perhaps not aware of clang-format's [statementmacros](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#statementmacros) setting?

>Additionally, many modern IDE's complain about the lack of ; at the end of what is interpreted as a statement.

I've never seen that, could you give an example?